libraries/package-deps-hash/src/getRepoState.ts

```diff
@@ -270,6 +277,149 @@ export function applyWorkingTreeState(
+  gitPath: string | undefined,
+  args: string[],
+  currentWorkingDirectory: string,
+  stdin?: Readable
+): Promise<string> {
+  const spawnOptions: IExecutableSpawnOptions = {
+    currentWorkingDirectory,
+    stdio: ['pipe', 'pipe', 'pipe']
+  };
+
+  let stdout: string = '';
+  let stderr: string = '';
+
+  const proc: child_process.ChildProcess = Executable.spawn(gitPath || 'git', args, spawnOptions);
+  proc.stdout!.setEncoding('utf-8');
+  proc.stderr!.setEncoding('utf-8');
+
+  proc.stdout!.on('data', (chunk: string) => {
+    stdout += chunk.toString();
+  });
+  proc.stderr!.on('data', (chunk: string) => {
+    stderr += chunk.toString();
+  });
+
+  if (stdin) {
+    stdin.pipe(proc.stdin!);
+  }
+
+  const [status] = await once(proc, 'exit');
+  if (status !== 0) {
+    throw new Error(`git ${args[0]} exited with code ${status}: ${stderr}`);
+  }
+
+  return stdout;
+}
+
+/**
+ * Gets the object hashes for all files in the Git repo, combining the current commit with working tree state.
+ * Uses async operations and runs all primary Git calls in parallel.
+ * @param rootDirectory - The root directory of the Git repository
+ * @param additionalRelativePathsToHash - Root-relative file paths to have Git hash and include in the results
+ * @param gitPath - The path to the Git executable
+ * @beta
+ */
+export async function getRepoStateAsync(
+  rootDirectory: string,
+  additionalRelativePathsToHash?: string[],
+  gitPath?: string
+): Promise<Map<string, string>> {
+  const statePromise: Promise<IGitTreeState> = spawnGitAsync(
+    gitPath,
+    ['--no-optional-locks', 'ls-tree', '-r', '-z', '--full-name', 'HEAD', '--'],
+    rootDirectory
+  ).then(parseGitLsTree);
+  const locallyModifiedPromise: Promise<Map<string, boolean>> = spawnGitAsync(
+    gitPath,
+    ['--no-optional-locks', 'status', '-z', '-u', '--no-renames', '--ignore-submodules', '--'],
+    rootDirectory
+  ).then(parseGitStatus);
+
+  const hashPaths: string[] = [];
+  async function* getFilesToHash(): AsyncIterableIterator<string> {
+    if (additionalRelativePathsToHash) {
+      for (const file of additionalRelativePathsToHash) {
+        hashPaths.push(file);
+        yield `${file}\n`;
+      }
+    }
+
+    const [{ files }, locallyModified] = await Promise.all([statePromise, locallyModifiedPromise]);
+
+    for (const [filePath, exists] of locallyModified) {
+      if (exists) {
+        hashPaths.push(filePath);
+        yield `${filePath}\n`;
+      } else {
+        files.delete(filePath);
+      }
+    }
+  }
+
+  const hashObjectPromise: Promise<string> = spawnGitAsync(
+    gitPath,
+    ['--no-optional-locks', 'hash-object', '--stdin-paths'],
+    rootDirectory,
+    Readable.from(getFilesToHash(), {
+      encoding: 'utf-8',
+      objectMode: false,
+      autoDestroy: true
+    })
+  );
+
+  const [{ files, submodules }, hashObject] = await Promise.all([
+    statePromise,
+    hashObjectPromise,
+    locallyModifiedPromise
```

Comment:
```suggestion
    hashObjectPromise
```

You're not looking at the result of `locallyModifiedPromise`, and it's already awaited in the `getFilesToHash` call.

Reply:
Technically this means `locallyModifiedPromise` is a hard dependency of `hashObjectPromise`, but if it throws, it is easier to surface the error directly here.